Source/JavaScriptCore/runtime/RegExpPrototype.cpp

```diff
@@ -623,6 +624,73 @@ JSC_DEFINE_HOST_FUNCTION(regExpProtoFuncSplitFast, (JSGlobalObject* globalObject
     RETURN_IF_EXCEPTION(scope, { });
     ASSERT(!input->isNull());
 
+    // Fast path for newline splitting pattern: \r\n?|\n
+    if (regexp->specificPattern() == Yarr::SpecificPattern::Newlines) {
+        JSValue limitValue = callFrame->argument(1);
+        unsigned limit = limitValue.isUndefined() ? 0xFFFFFFFFu : limitValue.toUInt32(globalObject);
+        RETURN_IF_EXCEPTION(scope, { });
+
+        if (!limit)
+            RELEASE_AND_RETURN(scope, JSValue::encode(constructEmptyArray(globalObject, nullptr)));
+
+        unsigned inputSize = input->length();
+
+        if (!inputSize) {
+            JSArray* result = constructEmptyArray(globalObject, nullptr);
+            RETURN_IF_EXCEPTION(scope, { });
+            result->putDirectIndex(globalObject, 0, inputString);
```

Comment:
RegExp matching is also changing RegExpCache state. So you need to make sure the cache state (`RegExp.$1` etc.) is the same to when we are doing matching.